docs/proposals/api-grouping.md

```diff
@@ -0,0 +1,427 @@
+	job
+
+/apis/longrunning/v1  # need better name
+	replicaSet
+	deployment
+	petSet
+
+/apis/cluster/v1
+	daemonSet
+
+# ReplicationController is deprecated by ReplicaSet.
+```
+
+Rationale for `batch` group is that if you are running a batch job, you are likely to care about
+all three things (run-to-completion, start-at-a-time, and dependencies).
+
+Rationale for `DaemonSet` being in separate group is that typically only the cluster admin
+needs to deal with DaemonSets.
+
+Other *verticals*, to use the Omega term, would be peers of these groupings.
+For example, a map-reduce controller might have a `MapReduce` object that specifies
+the data and program, and then it might make its own Pods.  How it negotiates
+for resources is a little unclear, though.
+
+### Horizontal Grouping
+
+Group things that are complementary, and have a similar control loop and/or similar level of abstraction.
+For example, put `Job`, `DaemonSet`, `ReplicaSet`, `Deployment` and `ReplicationController` all go into a group
+because they all make Pods.
+
+
+- pro: we have three similar things.  let us group them.  yay!
+- con: what about `PetSet`?
+- con: conflicts with *vertical* way of grouping things, and with *user vs admin*.
+
+### Respecting Dependencies
+
+If Object A1 in api group A strongly depends on Object B1 in api group B,
+and if api group A is "optional" then B should be "optional".
+But what does "strongly depend" mean?
+
+If Object C1 and Object C2 have a circular dependency, then it suggets that they belong in the same
+API group.
+
+### Conclusion
+
+The "user vs admin" has a clear benefit for hosting providers in that they can chose to not expose
+a single group which contains a bunch of things they do not want to expose (a custom scheduler
+might get nodes information from a different API).
+
+The "Vertical" grouping seems like it fits well with "conways law" and that "vertical" is the pattern that
+people who build large extensions will tend to follow.
+
+Therefore, I suggest a combination of the "user vs admin" and the "vertical" groupings.
+
+Sketch of final groupings (not showing types that may remain for deprecation purposes):
+
+```
+/api/v1
+	persistentVolumeClaim
+	pod
+	podTemplate
+	service
+	serviceAccount
+	endpoints
+	event
+	secret
+
+/apis/componentsStatus
+	componentStatus
+
+/apis/ingress
+	ingress
+
+/apis/metrics/v1alpha1
+	Pod
+
+/apis/cluster/v1
+	node
+	namespace
+	binding
+	persistentVolume
+	limitRange
+	resourceQuota
+	daemonSet
+	clusterAutoscaler
+	metadataPolicy
+	podSecurityPolicy
+
+/apis/clusterAutoScaling/v1
+	clusterAutoscaler
+
+/apis/scaling/v1
+	scale
+	horizontalPodAutoscaler
+
+/apis/thirdParty
+	thirdPartyResource
+	thirdPartyResourceData
+
+/apis/batch/v1
+	scheduledJob
+	workflow
+	job
+
+/apis/longrunning/v1  # need better name
+	replicaSet
+	deployment
+	petSet
+```
+
+
+Suggested order of work:
+
+- `Job` to move to `batch` for 1.2.
+- `ScheduledJob` and `Workflow` to move to `batch` when the become `v1`.
```

Comment:
SGTM my only question is what or rather where newly creatd objects land. Since you're proposing group's layout how about adding a sentence about naming scheme for new object that will be written soon, ScheduledJob to name one. This should provide enough guidance in the topic of API structuring.